// hedron/lib/components/Column.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  display: block;\n  ', '\n  box-sizing: border-box;\n  ', '\n  width: 100%;\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n'], ['\n  display: block;\n  ', '\n  box-sizing: border-box;\n  ', '\n  width: 100%;\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n    ', '\n    ', '\n  '], ['\n    ', '\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Row = require('./Row');

var _Row2 = _interopRequireDefault(_Row);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /* eslint-disable no-unused-vars */


function ColumnContainer(props) {
  var children = props.children,
      tagName = props.tagName,
      debug = props.debug,
      divisions = props.divisions,
      fluid = props.fluid,
      xs = props.xs,
      sm = props.sm,
      md = props.md,
      lg = props.lg,
      xsShift = props.xsShift,
      smShift = props.smShift,
      mdShift = props.mdShift,
      lgShift = props.lgShift,
      rest = _objectWithoutProperties(props, ['children', 'tagName', 'debug', 'divisions', 'fluid', 'xs', 'sm', 'md', 'lg', 'xsShift', 'smShift', 'mdShift', 'lgShift']);

  var newChildren = (0, _utils.passOn)(children, [_Row2.default], function (child) {
    return {
      debug: typeof child.props.debug === 'undefined' ? debug : child.props.debug
    };
  });
  return _react2.default.createElement(tagName || 'div', rest, newChildren);
}

ColumnContainer.propTypes = {
  children: _react2.default.PropTypes.node,
  tagName: _react2.default.PropTypes.string,
  className: _react2.default.PropTypes.string,
  debug: _react2.default.PropTypes.bool,
  divisions: _react2.default.PropTypes.number,
  fluid: _react2.default.PropTypes.bool,
  xs: _react2.default.PropTypes.number,
  sm: _react2.default.PropTypes.number,
  md: _react2.default.PropTypes.number,
  lg: _react2.default.PropTypes.number,
  xsShift: _react2.default.PropTypes.number,
  smShift: _react2.default.PropTypes.number,
  mdShift: _react2.default.PropTypes.number,
  lgShift: _react2.default.PropTypes.number
};

ColumnContainer.defaultProps = {
  divisions: 12
};

var Column = (0, _styledComponents2.default)(ColumnContainer)(_templateObject, function (props) {
  return props.debug ? 'background-color: rgba(50, 50, 255, .1);\n  border: 1px solid #fff;' : '';
}, function (props) {
  return props.fluid ? 'padding: 0;' : 'padding: 20px;';
}, function (props) {
  return props.xs ? 'width: ' + (0, _utils.divvy)(props.divisions, props.xs) + '%;' : null;
}, function (props) {
  return props.xsShift ? 'margin-left: ' + (0, _utils.divvy)(props.divisions, props.xsShift) + '%;' : null;
}, _utils.media.sm(_templateObject2, function (props) {
  return props.sm ? 'width: ' + (0, _utils.divvy)(props.divisions, props.sm) + '%;' : null;
}, function (props) {
  return props.smShift ? 'margin-left: ' + (0, _utils.divvy)(props.divisions, props.smShift) + '%;' : null;
}), _utils.media.md(_templateObject2, function (props) {
  return props.md ? 'width: ' + (0, _utils.divvy)(props.divisions, props.md) + '%;' : null;
}, function (props) {
  return props.mdShift ? 'margin-left: ' + (0, _utils.divvy)(props.divisions, props.mdShift) + '%;' : null;
}), _utils.media.lg(_templateObject2, function (props) {
  return props.lg ? 'width: ' + (0, _utils.divvy)(props.divisions, props.lg) + '%;' : null;
}, function (props) {
  return props.lgShift ? 'margin-left: ' + (0, _utils.divvy)(props.divisions, props.lgShift) + '%;' : null;
}));

exports.default = Column;
