// hedron/lib/components/Row.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  display: flex;\n  flex-direction: row;\n  flex-wrap: wrap;\n'], ['\n  display: flex;\n  flex-direction: row;\n  flex-wrap: wrap;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Column = require('./Column');

var _Column2 = _interopRequireDefault(_Column);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /* eslint-disable no-unused-vars */


function RowContainer(props) {
  var children = props.children,
      tagName = props.tagName,
      debug = props.debug,
      divisions = props.divisions,
      rest = _objectWithoutProperties(props, ['children', 'tagName', 'debug', 'divisions']);

  var newChildren = (0, _utils.passOn)(children, [_Column2.default], function (child) {
    return {
      debug: typeof child.props.debug === 'undefined' ? debug : child.props.debug,
      divisions: divisions
    };
  });
  return _react2.default.createElement(tagName || 'section', rest, newChildren);
}

RowContainer.propTypes = {
  children: _react2.default.PropTypes.node,
  tagName: _react2.default.PropTypes.string,
  className: _react2.default.PropTypes.string,
  debug: _react2.default.PropTypes.bool,
  divisions: _react2.default.PropTypes.number
};

RowContainer.defaultProps = {
  divisions: 12
};

var Row = (0, _styledComponents2.default)(RowContainer)(_templateObject);

exports.default = Row;
